### PR TITLE
Add jid 0.6.2 (New Formula)

### PR DESCRIPTION
**Pull request by ikeisuke**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
jid
https://github.com/simeji/jid

### CODE REVIEW

File: Formula/jid.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Jid < Formula
+  desc "jid"
```
Comment (neutric) [inline Formula/jid.rb]: Thanks for your contribution! A more comprehensive description of what jid does would be welcome.

File: Formula/jid.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  def install
+    name = Hardware::CPU.is_64_bit? ? "jid_darwin_amd64" : "jid_darwin_386"
+    mv "#{buildpath}/#{name}", "#{buildpath}/jid"
+    bin.install "jid"
+  end
+
+  test do
+    assert_match "jid version v#{version}", shell_output("#{bin}/jid --version")
```
Comment (neutric) [inline Formula/jid.rb]: A more complex test than a simple version string comparison is preferable.

Comment (neutric) [inline Formula/jid.rb]: A more complex test than a simple version string comparison is preferable.

File: Formula/jid.rb (hunk 1 above)
Comment (ikeisuke) [inline Formula/jid.rb]: Hi, @neutric. Thank you for review.
I've updated description.

File: Formula/jid.rb (hunk 2 above)
Comment (MikeMcQuaid) [inline Formula/jid.rb]: No, we'd rather use `assert_match` and `(testpath/"test.json").write` with some sample JSON for a better test.

File: Formula/jid.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class Jid < Formula
+  desc "Json incremental digger"
+  homepage "https://github.com/simeji/jid"
+  version "0.6.2"
+  if Hardware::CPU.is_64_bit?
+    url "#{homepage}/releases/download/#{version}/jid_darwin_amd64.zip"
```
Comment (MikeMcQuaid) [inline Formula/jid.rb]: This needs to be build from source instead.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/jid.rb (hunk 3 above)
Comment (ikeisuke) [inline Formula/jid.rb]: Hi @MikeMcQuaid . Thank you for review.
I've updated PR.

File: Formula/jid.rb (hunk 1 above)
Comment (MikeMcQuaid) [inline Formula/jid.rb]: Not seeing an updated description.

File: Formula/jid.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+class Jid < Formula
+  desc "jid"
+  homepage "https://github.com/simeji/jid"
+  version "0.6.2"
```
Comment (MikeMcQuaid) [inline Formula/jid.rb]: Remove this.

File: Formula/jid.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  url "#{homepage}/archive/#{version}.zip"
```
Comment (MikeMcQuaid) [inline Formula/jid.rb]: Hardcode this, don't use variables.

File: Formula/jid.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    (buildpath/"src/github.com/simeji/jid").install buildpath.children
+    cd buildpath/"src/github.com/simeji/jid" do
+      system "go", "get"
+      system "go", "build", "-o", bin/"jid", "cmd/jid/jid.go"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    assert_match "jid version v#{version}", shell_output("#{bin}/jid --version")
```
Comment (MikeMcQuaid) [inline Formula/jid.rb]: Still need a better test.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by ikeisuke** [comment]
@MikeMcQuaid thank you for review.

I've updated desc, removed version and change to hardcode url.

I'm considering a method of test code implementation.
I think that it is difficult to write a complex test unless update jid.

File: Formula/jid.rb (hunk 6 above)
Comment (MikeMcQuaid) [inline Formula/jid.rb]: Actually, this is fine 👍

**Review by MikeMcQuaid — APPROVED** [review]

File: Formula/jid.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
+class Jid < Formula
+  desc "Json incremental digger"
+  homepage "https://github.com/simeji/jid"
+  url "https://github.com/simeji/jid/archive/0.6.2.zip"
```
Comment (MikeMcQuaid) [inline Formula/jid.rb]: `https://github.com/simeji/jid/archive/0.6.2.tar.gz`

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format is `jid 0.6.2 (new formula)`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md). Thanks!

**Comment by neutric** [comment]
Thanks guys!

**Comment by ikeisuke** [comment]
@neutric, @MikeMcQuaid 
Thanks again for your help and your review here.

**Pull request closed without merging**